-1

**Failed Tests**:   Build 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-1d0c22/14303/summary.html
**COMMIT**: 45f5becbd0d603f7fa4d7473df54bc0e52307e0d
**CMSSW**: CMSSW_12_0_X_2021-04-18-0000/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/33458/14303/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-04-18-0000/src/RecoEcal/EgammaClusterProducers/src/ReducedRecHitCollectionProducer.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-04-18-0000/src/RecoEcal/EgammaClusterProducers/src/SCEnergyCorrectorProducer.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-04-18-0000/src/RecoEcal/EgammaClusterProducers/src/SealModule.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-04-18-0000/src/RecoEcal/EgammaClusterProducers/src/UncleanSCRecoveryProducer.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-04-18-0000/src/RecoEcal/EgammaClusterProducers/src/UnifiedSCCollectionProducer.cc
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-04-18-0000/src/RecoEcal/EgammaClusterProducers/src/SCEnergyCorrectorProducer.cc:28:32: error: 'SuperClusterCollection' is not a member of 'reco'; did you mean 'CaloClusterCollection'?
   28 |   const edm::EDGetTokenT<reco::SuperClusterCollection> inputSCToken_;
      |                                ^~~~~~~~~~~~~~~~~~~~~~
      |                                CaloClusterCollection
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-04-18-0000/src/RecoEcal/EgammaClusterProducers/src/SCEnergyCorrectorProducer.cc:28:54: error: template argument 1 is invalid
   28 |   const edm::EDGetTokenT<reco::SuperClusterCollection> inputSCToken_;
</pre>

--- a/RecoEcal/EgammaClusterProducers/src/SCEnergyCorrectorProducer.cc
+++ b/RecoEcal/EgammaClusterProducers/src/SCEnergyCorrectorProducer.cc
@@ -4,6 +4,7 @@
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 
 #include "DataFormats/EgammaReco/interface/SuperCluster.h"
+#include "DataFormats/EgammaReco/interface/SuperClusterFwd.h"
 #include "DataFormats/Common/interface/ValueMap.h"
 #include "RecoEcal/EgammaClusterAlgos/interface/SCEnergyCorrectorSemiParm.h"
 

